### PR TITLE
Firerate drop enemies

### PR DESCRIPTION
**Pull request by whitew1994WW**
All enemies drop an object that will be a firerate boost. The boost objects move slowly towards you. These boosts are not permanent and the more you take damage the more of the boosts you loose.

Currently doesn't boost the firerate, just the object complete.

### CODE REVIEW

**Comment by whitew1994WW** [comment]
This PR also includes partial changes to the HUD. The rest of these changes will come in a separate PR, including centring the map and resizing the player view screes so that the player isn't hidden by the HUD

**Comment by Jamesbarford** [comment]
Sorry will, I think the changes I merged have massively clashed here. The positive is that this should now be achievable in one place 🤞 (I hope)

File: Moonshot/Combat/Stats.gd [hunk 1]
```diff
@@ -7,12 +7,17 @@ class_name Stats
...
+var modifiers = {"firerate_pickups": 0, "firerate": 1}
+var no_firerate_pickups_to_increase_firerate = 2
+var firerate_scaling_factor = 1.2
+var max_firerate_level = 5
+var current_firerate_level = 0
 
 var invulnerable := false
 
-export var max_health := 1.0 setget set_max_health, get_max_health
+export var max_health := 0.2 setget set_max_health, get_max_health
```
Comment (Jamesbarford) [inline Moonshot/Combat/Stats.gd:20]: Set this back to 1?

File: Moonshot/items_objects/FireRatePickup.gd [hunk 2]
```diff
@@ -0,0 +1,33 @@
+extends RigidBody2D
+
+# Declare member variables here. Examples:
+# var a = 2
+# var b = "text"
+
+var shot_speed = 0.01
```
Comment (Jamesbarford) [inline Moonshot/items_objects/FireRatePickup.gd:7]: is this the speed of the shot or the speed of the pickup?

it seems to be the speed of which the pickup travels toward the player, maybe `const PICK_UP_TRAVEL_SPEED`

File: Moonshot/items_objects/FireRatePickup.gd [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+#func _process(delta):
+#	pass
+
+
+
+func _on_FirerateDrop_body_entered(body):
+	if body.is_in_group('Player'):
+		Utils.player_stats.pickup_firerate(1)
```
Comment (Jamesbarford) [inline Moonshot/items_objects/FireRatePickup.gd:31]: is one the shot speed increase?

maybe have a `const FIRE_RATE_INCREASE =. 1`

**Review by Jamesbarford — CHANGES_REQUESTED** [review]

File: Moonshot/items_objects/FireRatePickup.gd (hunk 3 above)
Comment (whitew1994WW) [reply Moonshot/items_objects/FireRatePickup.gd:31]: Yeah can't hurt

File: Moonshot/items_objects/FireRatePickup.gd (hunk 2 above)
Comment (whitew1994WW) [reply Moonshot/items_objects/FireRatePickup.gd:7]: Yep will change this too

**Pull request merged**